src/mame/jaleco/megasys1.cpp

```diff
@@ -398,6 +453,103 @@ void megasys1_state::megasys1C_map(address_map &map)
 	map(0x1c0000, 0x1cffff).mirror(0x30000).ram().w(FUNC(megasys1_state::ram_w)).share("ram"); //0x1f****, Cybattler reads attract mode inputs at 0x1d****
 }
 
+void megasys1_bc_iosim_state::megasys1C_iosim_map(address_map &map)
+{
+	megasys1C_map(map);
+	map(0x0d8000, 0x0d8001).rw(FUNC(megasys1_bc_iosim_state::ip_select_r), FUNC(megasys1_bc_iosim_state::ip_select_w));
+}
+
+/*
+  Used by IO MCU to read data from different inputs (DSW1, DSW2, P1, P2, COIN) depending on
+  which one was previously selected by the MCU itself using the offset.
+  It also sends the IRQ2 to the main CPU for some case
+*/
+u8 megasys1_bc_iomcu_state::mcu_capture_inputs_r(offs_t offset)
+{
+	u8 input_data = 0x00;
+	u8 bank = offset >> 16;
+
+	switch (bank)
+	{
+		case 0x01:  input_data = m_io_p1->read(); break;
+		case 0x02:  input_data = m_io_p2->read(); break;
+		case 0x03:  input_data = m_io_dsw1->read(); break;
+		case 0x04:  input_data = m_io_dsw2->read(); break;
+		case 0x05:  input_data = m_io_system->read(); break;
+		case 0x07:
+			m_maincpu->set_input_line(2, HOLD_LINE);
+			LOG("%s: mcu_capture_inputs_r: Send IRQ2 to main CPU (address %06x)\n", machine().describe_context(), offset);
+			return input_data;
+		default:    LOG("%s: mcu_capture_inputs_r: Invalid input selected (data %02x)\n", machine().describe_context(), bank); return input_data;
+	}
+
+	LOG("%s: mcu_capture_inputs_r: Read data from inputs: (data %02x)\n", machine().describe_context(), input_data);
+
+	return input_data;
+}
+
+/*
+  Used by IO MCU to read data from the main CPU when INT0 is triggered
+*/
+u8 megasys1_bc_iomcu_state::mcu_port1_r()
+{
+	LOG("%s: mcu_port1_r: Read data from CPU: (data %02x)\n", machine().describe_context(), m_mcu_input_data);
+
+	return m_mcu_input_data;
+}
+
+/*
+  Used by IO MCU to send data to the main CPU when INT0 is triggered
+*/
+void megasys1_bc_iomcu_state::mcu_port2_w(u8 data)
+{
+	LOG("%s: mcu_port2_w: Send data to CPU: (data %02x)\n", machine().describe_context(), data);
+
+	m_mcu_io_data = data;
+}
+
+/*
+  Unknown purpose, but the IO MCU writes into the bit 3 of the port 6
+*/
+void megasys1_bc_iomcu_state::mcu_port6_w(u8 data)
+{
+	LOG("%s: mcu_port6_w: Write data to port 6: (data %02x)\n", machine().describe_context(), data);
+}
+
+/*
+  Read handler triggered by the main CPU at some specific address. Used to read data from port 2 of IO MCU
+*/
+u16 megasys1_bc_iomcu_state::ip_select_bigstrik_r() // FROM MCU
+{
+	LOG("%s: ip_select_bigstrik_r: Read data from MCU: (data %02x)\n", machine().describe_context(), m_mcu_io_data);
+
+	return m_mcu_io_data;
+}
+
+/*
+  Write handler triggered by the main CPU at some specific address. Used to write data to port 1 of IO MCU and send irq0
+*/
+void megasys1_bc_iomcu_state::ip_select_bigstrik_w(u16 data) // TO MCU
+{
+	LOG("%s: ip_select_bigstrik_w: Send data to MCU: (data %02x)\n", machine().describe_context(), data);
+
+	m_mcu_input_data = data;
+
+	m_iomcu->set_input_line(INPUT_LINE_IRQ0, ASSERT_LINE);
```

Comment:
Similarly, where is this being deasserted?

Reply:
We can change it to `HOLD_LINE`.

Reply:
In this case, the IRQ0 line is only active during the time spent by the CPU to write the data for the MCU, i.e. only a CPU write cycle.
Then, I don't know how to implement that at code level and where to place the `CLEAR_LINE`.
Does we need to use `pulse_input_line` in this case? How to set the proper lenght of the generated pulse?

Reply:
Yes, you can use `pulse_input_line` in that case.  If it’s for a single bus cycle, you can use `attotime::from_ticks(1, m_maincpu->clock())` or something as the duration.